src/_guides/language/extension-methods.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Extension methods
+description: Learn how to add to existing APIs.
+---
+<?code-excerpt replace="/ *\/\/\s+ignore_for_file:[^\n]+\n//g; /([A-Z]\w*)\d\b/$1/g; /\b(main)\d\b/$1/g"?>
+
+Extension methods, introduced in Dart 2.7,
+are a way to add functionality to existing libraries.
+You might not ever need to _use_ extension methods, much less _create_ them.
+But if you generate code or find yourself writing lots of wrappers,
+consider using extension methods to reduce
+the amount of boilerplate code that you need to write.
+
+You might use extension methods without even knowing it.
+For example, when you use code completion in an IDE,
+it suggests extension methods alongside regular methods.
+
+{% comment %}
+[PENDING: embed video here.]
+Watch this video for an introduction to extension methods.
+{% endcomment %}
+
+## Overview
+
+When you’re using someone else’s API or
+when you implement a library that’s widely used,
+it’s often impractical or impossible to change the API.
+But you might still want to add some functionality.
+
+For example, consider the following code that parses a string into an integer:
+
+```dart
+int.parse('42')
+```
+
+It might be nice — shorter and easier to use with tools — to
+have that functionality be in the string instead:
+
+```dart
+'42'.toInt()
+```
+
+To enable that code,
+you can import a library that contains an extension of the String class:
+
+```dart
+import string_apis.dart';
+...
+print('42'.toInt());
+```
+
+Extensions can define not just methods and getters,
```

Comment:
The grouping of methods + getters vs setters + operators in other is a bit surprising. Maybe change to "Extensions can define not just methods and getters,
 

 
The grouping of methods + getters vs setters + operators in other is a bit surprising. Maybe change to "Extensions can define not just methods, but also other members such as getters, setters, and operators." ?

Reply:
Yep. This is a leftover from old code that used a getter, not an extension method.